repository.go

```diff
@@ -196,15 +199,25 @@ func PlainOpen(path string) (*Repository, error) {
 	var wt, dot billy.Filesystem
 
 	fs := osfs.New(path)
-	if _, err := fs.Stat(".git"); err != nil {
+	fi, err := fs.Stat(".git")
+	if err != nil {
```

Comment:
I think the logic here is correct, but we can improve its readability by hiding the low-level details: add a switch with few functions, like `isWorkingDirectory`, `isSubModule`, `isBare`... that reveal our intentions to the reader, hiding the details, that are sometimes unkown to the reader and much more difficult to get right).

This will also fix @mcuadros comment as a bonus.